### PR TITLE
Add manubot webpage subcommand

### PR DESCRIPTION
**Pull request by dhimmel**
Move the manubot webpage command [from rootstock](https://github.com/manubot/rootstock/blob/bc094bd3d6cec6935298b4acf50e45ad55bf45c8/build/webpage.py) to a subcommand of the python CLI.

Makes some unrelated changes to `setup.py` to keep with the times.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ build: off
 install:
   - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
   - "python -m pip install --upgrade pip"
-  - "%CMD_IN_ENV% pip install ."
+  - "%CMD_IN_ENV% pip install .[all]"
 
 before_test:
   - "%CMD_IN_ENV% pip install pytest"
```
Comment (agitter) [inline .appveyor.yml]: Can this line be removed now?  The "all" feature installs pytest.

File: setup.py [hunk 2]
```diff
@@ -15,12 +16,29 @@
...
+# extra depedencies with an "all" option
+extras_require = {
+    'webpage': [
+        'opentimestamps-client',
+    ],
+    'test': [
+        'pytest',
+    ],
+}
+extras_require['all'] = list(dict.fromkeys(itertools.chain.from_iterable(extras_require.values())))
```
[the remark this answers is not in the document]
Comment (dhimmel) [reply setup.py]: > Is it correct that all isn't a special keyword here?

Yes you are correct. PEP 426 [proposed](https://www.python.org/dev/peps/pep-0426/#extras-optional-dependencies) and `all` keyword for installing all extras. But that PEP was withdrawn, so this just borrows the proposed syntax and explicitly implements the functionality.

**Comment by dhimmel** [comment]
> Were there any major changes to the former webpage.py behavior that I should review carefully? I didn't notice any.

Minimal changes to the behavior like that we now create the webpage directory if it doesn't exist. Maybe one other place where I changed the behavior to be a bit more robust, but the default workflow should be the same.

**Comment by agitter** [comment]
Thanks.  This looks ready to merge.

**Comment by dhimmel** [comment]
Somehow https://github.com/manubot/manubot/pull/132/commits/4a14767418a60e4595b5cf9aae8fb50a608e5ffa got added to the master branch... not sure how:

image

**Comment by dhimmel** [comment]
> Will email github support

Stacey Burns from GitHub Developer Support got back to me with the following message:

> When merging the Pull Request on the UI, do you use ctrl+enter/cmd+enter to submit the change? We have an open bug report where merge commits are created instead of squash merge in some cases when using keyboard shortcuts.
>
> I'm wondering if this might be a result of this bug.

I responded:

> I don't use the "ctrl+enter/cmd+enter" keyboard shortcuts, but it's possible I accidentally triggered one of those shortcuts, which performed a merge. Before the page updated, perhaps then I also clicked the squash merge button resulting in the "the head tip has changed" error message. I'm not 100% sure this was the issue, but it seems plausible. Thanks for looking into this.

**Pull request merged**